frontend/javascripts/oxalis/controller/combinations/skeletontracing_plane_controller.js

```diff
@@ -144,6 +146,74 @@ export function moveNode(dx: number, dy: number) {
   );
 }
 
+function otherNodeOfEdge(edge: Edge, nodeId: number): number {
+  return edge.source === nodeId ? edge.target : edge.source;
+}
+function getSubsequentNodeFromTree(tree: Tree, node: Node): number {
+  const nodes = tree.edges.getEdgesForNode(node.id).map(edge => otherNodeOfEdge(edge, node.id));
+  const next = Math.max(node.id, ...nodes);
```

Comment:
Why do you put `node.id` in there as well? I would say only the node ids of the neighbors need to be checked. Otherwise users will get stuck in places where the current node has a higher id than all of its neighbors. (Same for the `getPrecedingNodeFromTree` function)

Reply:
Otherwise there can be loops. For example if we have three nodes (1, 2, 3) and 2 is active. Then we move forward, so 3 is active and the list is [2, 3]. Now we move to the next again. Because there is no next in the list we search for it in the tree. If we don't consider 3 we move to 2 and have the list [2, 3, 2]. Moving forward again extends this even further. And when moving back you would jump between nodes 2 and 3 several times before arriving at node 1.
So both ways lead to problems.

Reply:
What about checking what the last node was (depending on the direction the user is going with the shortcuts) by looking it up in the list (if one exists) and then making sure to exclude that node when looking for the new node to travel to? Does that make sense?

Reply:
I think that should work :)